examples/Catena4430_Sensor/Catena4430_cMeasurementLoop_SDcard.cpp

```diff
@@ -571,4 +575,75 @@ cMeasurementLoop::updateFromSd(
         }
     }
 
+void
+cMeasurementLoop::handleSdTTNv3Migrate(
+    void
+    )
+    {
+    static const char * const sMigrate = "MIGRATE.V3";
+    bool fMigrate;
+    bool fResult = this->checkSdCard();
+
+	if (fResult)
+        {
+        fMigrate = this->handleSdTTNv3MigrateCardUp(sMigrate);
+        }
+
+    if (fMigrate)
+        {
+        bool fFramUpdate = this->updateFramAppEui();
+
+        if (fFramUpdate)
+            {
+            this->reJoinNetwork();
+            gSD.remove(sMigrate);
+            gLog.printf(gLog.kInfo, "cFramStorage::kAppEUI: update: success\n");
+			}
+        else
+            gLog.printf(gLog.kError, "cFramStorage::kAppEUI: not updated\n");
+        }
+    }
+
+bool
+cMeasurementLoop::handleSdTTNv3MigrateCardUp(
+    const char *sMigrate
+    )
+    {
+    if (! gSD.exists(sMigrate))
+        {
+        if (gLog.isEnabled(gLog.kTrace))
+            gLog.printf(gLog.kAlways, "%s: not found: %s\n", FUNCTION, sMigrate);
+
+        return false;
+        }
+    return true;
+    }
+
+bool
+cMeasurementLoop::updateFramAppEui(
+    void
+    )
+    {
+    auto const pFram = gCatena.getFram();
+    uint8_t AppEUI[] = { 1, 0, 0, 0, 0, 0, 0, 0 };
+
+    if (pFram == nullptr)
+        return false;
+
+    pFram->saveField(cFramStorage::kAppEUI, AppEUI);
```

Comment:
Check error status.

Reply:
yes, I have planned to check return value, unfortunately API [saveField](https://github.com/mcci-catena/Catena-Arduino-Platform/blob/6c1db300e2fca6bf7ed430f58e2598ad4c4f65cf/src/Catena_Fram.h#L123-L129) is of datatype `void`.

In this case, I believe we can use API [getField](https://github.com/mcci-catena/Catena-Arduino-Platform/blob/6c1db300e2fca6bf7ed430f58e2598ad4c4f65cf/src/Catena_Fram.h#L138-L144) to read the value written in FRAM, and then compare to check for error.

Please advice on this

Reply:
See previous.